src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs

```diff
@@ -0,0 +1,152 @@
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.VisualStudio.Services.WebPlatform;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Telemetry
+{
+    public class TelemetryCommandExtension : AgentService, IWorkerCommandExtension
+    {
+        private string _area;
+        private string _feature;
+        public HostTypes SupportedHostTypes => HostTypes.Build;
+
+        public void ProcessCommand(IExecutionContext context, Command command)
+        {
+            if (string.Equals(command.Event, WellKnownEventTrackCommand.Publish, StringComparison.OrdinalIgnoreCase))
+            {
+                ProcessPublishTelemetryCommand(context, command.Properties, command.Data);
+            }
+            else
+            {
+                throw new Exception(StringUtil.Loc("TelemetryCommandNotFound", command.Event));
+            }
+        }
+
+        public Type ExtensionType
+        {
+            get
+            {
+                return typeof(IWorkerCommandExtension);
+            }
+        }
+
+        public string CommandArea
+        {
+            get
+            {
+                return "telemetry";
+            }
+        }
+
+        private void ProcessPublishTelemetryCommand(IExecutionContext context, Dictionary<string, string> eventProperties, string data)
+        {
+            ArgUtil.NotNull(context, nameof(context));
+
+            LoadTelemetryInputs(eventProperties);
+
+            var ciService = HostContext.GetService<ICustomerIntelligenceServer>();
+            var vssConnection = WorkerUtilies.GetVssConnection(context);
+            ciService.Initialize(vssConnection);
+
+            var commandContext = HostContext.CreateService<IAsyncCommandContext>();
+            commandContext.InitializeCommandContext(context, StringUtil.Loc("Telemetry"));
+            commandContext.Task = ciService.PublishEventsAsync(new CustomerIntelligenceEvent[] { PopulateCustomerIntelligenceData(context, data) });
```

Comment:
have you try the behavior when you get tarpit?
https://vsowiki.com/index.php?title=Resource_Utilization_Service